### PR TITLE
use configGenerator

### PR DESCRIPTION
**Pull request by rickardenglund**
Use ConfigGenerator instead of GetKeyFunc

### CODE REVIEW

File: v2/http-middleware/ratelimit/ratelimit.go [hunk 1]
```diff
@@ -52,7 +51,7 @@ func CreateLimiter(s Store) Limiter {
 // If you give multiple configs for 1 endpoint. The most restrictive one will apply
 // The algorithm is inspired from: https://redislabs.com/redis-best-practices/basic-rate-limiting/
 func (s *Limiter) Configure(config EndpointConfig) {
```
Comment (Ekberg) [inline v2/http-middleware/ratelimit/ratelimit.go:53]: Here we could skip the EndointConfig struct and just use two arguments? Maybe add a type for the `func(*http.Request) ([]Config, error){}`

Comment (Ekberg) [reply v2/http-middleware/ratelimit/ratelimit.go:53]: and use `time.Minute `to make it more clear what we are doing.

Comment (rickardenglund) [reply v2/http-middleware/ratelimit/ratelimit.go]: Where should I use time.Minute? the requests per minute is not minutes?

**Comment by j0s** [comment]
This comment is not really related to this PR, but i post it here anyway since you are doing related changes.

Many endpoints contain a user ID or a node ID in its path, can this solution handle that? Shouldn't Request.Path be a regular expression?

**Review by Ekberg — APPROVED** [review]

**Pull request merged**